samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -280,6 +282,7 @@ class TaskInstance(
       if (timeSinceLastCommit < commitMaxDelayMs) {
         info("Skipping commit for taskName: %s since another commit is in progress. " +
           "%s ms have elapsed since the pending commit started." format (taskName, timeSinceLastCommit))
+        metrics.asyncCommitSkipped.set(numSkippedCommits + 1)
```

Comment:
This is always 1. You probably need to do `metrics.asyncCommitSkipped.set(metrics.asyncCommitSkipped.get + 1)`.

Reply:
Also the metric name should just be commitsSkipped / skippedCommits.